spec/ios_generate_strings_file_from_code_spec.rb

```diff
@@ -0,0 +1,67 @@
+  let(:test_data_dir) { File.join(File.dirname(__FILE__), 'test-data', 'translations', 'ios_generate_strings_file_from_code') }
+  let(:app_src_dir) { File.join(test_data_dir, 'sample-project', 'Sources') }
+  let(:pod_src_dir) { File.join(test_data_dir, 'sample-project', 'Pods') }
+
+  def test_genstrings(paths_to_scan:, quiet:, swiftui:, expected_dir_name:, expected_logs: nil)
+    Dir.mktmpdir('a8c-wpmrt-ios_generate_strings_file_from_code-') do |tmp_dir|
+      # Act
+      genstrings_logs = described_class.run(paths: paths_to_scan, quiet: quiet, swiftui: swiftui, output_dir: tmp_dir)
+
+      output_files = Dir[File.join(tmp_dir, '*.strings')]
+      expected_files = Dir[File.join(test_data_dir, expected_dir_name, '*.strings')]
+
+      # Assert: same list of generated files
+      expect(genstrings_logs).to eq(expected_logs) unless expected_logs.nil?
+      expect(output_files.map { |f| File.basename(f) }.sort).to eq(expected_files.map { |f| File.basename(f) }.sort)
+
+      # Assert: each generated file has expected content
+      output_files.each do |generated_file|
+        file_basename = File.basename(generated_file)
+        expected_file = expected_files.find { |f| File.basename(f) == file_basename }
+        expect(File.read(generated_file)).to eq(File.read(expected_file)), "Content of '#{file_basename}' and '#{expected_file}' do not match."
+      end
+    end
+  end
+
+  context 'when including pods' do
+    it 'Generates the expected .strings files with SwiftUI support' do
+      test_genstrings(paths_to_scan: [app_src_dir, pod_src_dir], quiet: true, swiftui: true, expected_dir_name: 'expected-pods-swiftui')
+    end
+
+    it 'Generates the expected .strings files without SwiftUI support' do
+      test_genstrings(paths_to_scan: [app_src_dir, pod_src_dir], quiet: true, swiftui: false, expected_dir_name: 'expected-pods-noswiftui')
+    end
+  end
+
+  context 'when not including pods' do
+    it 'Generates the expected .strings files with SwiftUI support' do
+      test_genstrings(paths_to_scan: [app_src_dir], quiet: true, swiftui: true, expected_dir_name: 'expected-nopods-swiftui')
+    end
+
+    it 'Generates the expected .strings files without SwiftUI support' do
+      test_genstrings(paths_to_scan: [app_src_dir], quiet: true, swiftui: false, expected_dir_name: 'expected-nopods-noswiftui')
+    end
+  end
+
+  context 'when genstrings finds warnings' do
+    it 'only logs warnings about multiple values in quiet mode' do
+      expected_logs = [
+        %(Key "app.key5" used with multiple values. Value "app value 5\\nwith multiple lines." kept. Value "app value 5\\nwith multiple lines, and different value than in Swift" ignored.),
+      ]
+      test_genstrings(paths_to_scan: [app_src_dir, pod_src_dir], quiet: true, swiftui: true, expected_dir_name: 'expected-pods-swiftui', expected_logs: expected_logs)
+    end
+
+    it 'logs warnings about both duplicate values and duplicate comments if not in quiet mode' do
```

Comment:
Question. Have you considered using another `context` nesting?

```ruby
  context 'when genstrings finds warnings' do
    context 'when in quiet mode' do
      it 'only logs warnings about duplicate values' do
      end
    end

    context 'when not in quiet mode' do
      it 'logs warnings about both duplicate values and duplicate comments' do
```

I'm not sure how I feel about multiple `context` either. It does make the test structure clearer, but contributes to the pyramid of doom ⚖️ 🤷

Reply:
Yeah I went back and forth in how to structure that.

Similarly for the context nesting I used for Pods On/Off + SwiftUI On/Off, I am not sure how to split that nicely, given it's a 2x2 matrix of options to test, but we can only choose one axis when nesting in one `context` – or if we nest in 2 `context` levels, one per axis, we'll have to privilege one axis (e.g. Pods) over the other (e.g. SwiftUI).

Reply:
I've reorganized the test examples to be grouped differently and adapted the nesting accordingly, I feel it now makes more sense that way, but curious to hear what you think of the new spec structure.